components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts

```diff
@@ -211,53 +344,59 @@ export class AutomateSettingsComponent implements OnInit {
 
   // Update forms until we get the job scheduler status
   public updateForm(jobSchedulerStatus: JobSchedulerStatus) {
+
     if (jobSchedulerStatus === null) {
       return;
     }
 
     jobSchedulerStatus.jobs.forEach((job: IngestJob) => {
-      const [threshold, unit] = this.splitThreshold(job.threshold);
-      const form = {
-        disable: !job.running,
-        threshold: threshold,
-        unit: unit
-      };
 
-      switch (job.name) {
-        case IngestJobs.MissingNodes: {
-          this.missingNodesForm = this.formBuilder.group(form);
-          break;
+      switch (job.category) {
+        case 'infra': {
+          this.populateInfra(job);
         }
-        case IngestJobs.MissingNodesForDeletion: {
-          this.deleteMissingNodesForm = this.formBuilder.group(form);
-          break;
+        break;
+
+        case 'compliance': // fallthrough
+        case 'event_feed': {
+          this.populateNested(job);
         }
-        // TODO @afiune missing forms to add, at the moment we can't modify
-        // this parameter/settings since the services take it at startup.
-        // (we need to change that first)
-        //
-        // this.clientRunsForm = this.formBuilder.group(form);
-        // this.complianceDataForm = this.formBuilder.group(form);
-        // this.eventFeedForm = this.formBuilder.group(form);
-      }
-    });
+        break;
 
-    this.automateSettingsForm = this.formBuilder.group({
-      eventFeed: this.eventFeedForm,
-      clientRuns: this.clientRunsForm,
-      complianceData: this.complianceDataForm,
-      missingNodes: this.missingNodesForm,
-      deleteMissingNodes: this.deleteMissingNodesForm
+        default:
+          break;
+      }
     });
   }
 
-  private getJobForm(job: string) {
-    switch (job) {
-      case IngestJobs.MissingNodes: {
-        return this.automateSettingsForm.value['missingNodes'];
+  private getJobForm(jobName: string) {
```

Comment:
In case I was not clear, was just showing this as an intermediate step. 😁 Since you only call this function in one place, consider getting rid of it completely. Just use `this[jobName]` where it is called from.